### ISSUE TITLE
Does not parse relative URLs in Linked Data documents

### ISSUE THREAD

**Issue by RubenVerborgh** [comunica/comunica#311]
#### Issue type:

- :bug: Bug

____
#### Description:

```
$ comunica-sparql https://rubentest.solid.community/profile/card#me 'SELECT * WHERE { ?s ?p ?o }'
[
{"?s":"undefined./card","?p":"http://www.w3.org/1999/02/22-rdf-syntax-ns#type","?o":"http://xmlns.com/foaf/0.1/PersonalProfileDocument"},
{"?s":"undefined./card","?p":"http://xmlns.com/foaf/0.1/maker","?o":"null#me"},
{"?s":"undefined./card","?p":"http://xmlns.com/foaf/0.1/primaryTopic","?o":"null#me"},
{"?s":"null#me","?p":"http://www.w3.org/1999/02/22-rdf-syntax-ns#type","?o":"http://schema.org/Person"},
{"?s":"null#me","?p":"http://www.w3.org/1999/02/22-rdf-syntax-ns#type","?o":"http://xmlns.com/foaf/0.1/Person"},
{"?s":"null#me","?p":"http://www.w3.org/2006/vcard/ns#fn","?o":"\"Ruben Verborgh\""},
{"?s":"null#me","?p":"http://www.w3.org/ns/ldp#inbox","?o":"undefined/inbox/"},
{"?s":"null#me","?p":"http://www.w3.org/ns/pim/space#preferencesFile","?o":"undefined/settings/prefs.ttl"},
{"?s":"null#me","?p":"http://www.w3.org/ns/pim/space#storage","?o":"undefined/"},
{"?s":"null#me","?p":"http://www.w3.org/ns/solid/terms#account","?o":"undefined/"},
{"?s":"null#me","?p":"http://www.w3.org/ns/solid/terms#privateTypeIndex","?o":"undefined/settings/privateTypeIndex.ttl"},
{"?s":"null#me","?p":"http://www.w3.org/ns/solid/terms#publicTypeIndex","?o":"undefinedpublicTypeIndex.ttl"},
{"?s":"null#me","?p":"http://www.w3.org/ns/solid/terms#publicTypeIndex","?o":"undefined/settings/publicTypeIndex.ttl"},
{"?s":"null#me","?p":"http://xmlns.com/foaf/0.1/name","?o":"\"Ruben Verborgh (test)\""}
]
```

The problem is that https://rubentest.solid.community/profile/card#me contains

```turtle
@prefix : <#>.
```

and this relative URL is not resolved against the document base.
____
#### Environment:

| software            | version
| ------------------- | -------
| Comunica Init Actor | 1.4.1
| node                | v10.9.0
| npm                 | 6.4.1
| yarn                | 1.10.1
| Operating System    | darwin (Darwin 17.7.0)

**Comment by RubenVerborgh** [comment]
Seems like:
- [ ] `actor-rdf-parse-n3` should accept a `baseIRI` argument and pass it to `StreamParser`
- [ ] actors on the `bus-rdf-parse` should pass that argument
- [ ] RDF/XML parser needs to be checked
- [ ] JSON-LD parser needs to be checked
- [ ] should be tested to work with Linked Data documents as well as TPF (and any other response really)

**Comment by rubensworks** [comment]
Apparently, this has been fixed on the link traversal branch already (105eb681129356fd2c0a71fcecc2ec50d3a2cdae), so I'll cherry pick it from there. (This doesn't include RDF?XML parsing, so that will still need to be applied)

**Comment by RubenVerborgh** [comment]
Awesome, thanks!
